confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceInputFilterStream.java

```diff
@@ -700,41 +762,76 @@ private void readPageRevision(long pageId, String spaceKey, ConfluenceProperties
 
         // Content
-        if (bodyContent != null) {
-            if (this.properties.isContentEvents() && filter instanceof Listener) {
+        if (!pageProperties.containsKey(ConfluenceXMLPackage.KEY_PAGE_BLOGPOST)) {
+            if (bodyContent != null) {
+                if (this.properties.isContentEvents() && filter instanceof Listener) {
+                    // > WikiDocumentRevision
+                    proxyFilter.beginWikiDocumentRevision(revision, documentRevisionParameters);
+
+                    try {
+                        parse(bodyContent, bodyType, this.properties.getMacroContentSyntax(), proxyFilter);
+                    } catch (Exception e) {
+                        this.logger.warn("Failed to parse content of page with id [{}]. Cause: [{}].",
+                            createPageIdentifier(pageId, spaceKey), ExceptionUtils.getRootCauseMessage(e));
+                    }
+                } else if (this.properties.isConvertToXWiki()) {
+                    // Convert content to XWiki syntax
+                    try {
+                        documentRevisionParameters.put(WikiDocumentFilter.PARAMETER_CONTENT,
+                            convertToXWiki21(bodyContent, bodyType));
+                        documentRevisionParameters.put(WikiDocumentFilter.PARAMETER_SYNTAX, Syntax.XWIKI_2_1);
+                    } catch (Exception e) {
+                        this.logger.warn("Failed to convert content of the page with id [{}]. Cause: [{}].",
+                            createPageIdentifier(pageId, spaceKey), ExceptionUtils.getRootCauseMessage(e));
+                    }
+
+                    // > WikiDocumentRevision
+                    proxyFilter.beginWikiDocumentRevision(revision, documentRevisionParameters);
+                } else {
+                    // Keep Confluence syntax
+                    documentRevisionParameters.put(WikiDocumentFilter.PARAMETER_CONTENT, bodyContent);
+                    documentRevisionParameters.put(WikiDocumentFilter.PARAMETER_SYNTAX, bodySyntax);
+
+                    // > WikiDocumentRevision
+                    proxyFilter.beginWikiDocumentRevision(revision, documentRevisionParameters);
+                }
+            } else {
                 // > WikiDocumentRevision
                 proxyFilter.beginWikiDocumentRevision(revision, documentRevisionParameters);
+            }
+        } else {
+            String blogPostContent = bodyContent;
 
-                try {
-                    parse(bodyContent, bodyType, this.properties.getMacroContentSyntax(), proxyFilter);
-                } catch (Exception e) {
-                    this.logger.warn("Failed to parse content of page with id [{}]. Cause: [{}].",
-                        createPageIdentifier(pageId, spaceKey), ExceptionUtils.getRootCauseMessage(e));
-                }
-            } else if (this.properties.isConvertToXWiki()) {
+            if (bodyContent != null && this.properties.isConvertToXWiki()) {
```

Comment:
It's a bit strange to repeat `bodyContent != null` in each branch, seems to me you should put all that in a `if (bodyContent != null)`.